File: saraandeva/.claude/skills/saraandeva-episode/scripts/normalizeClipFilenames.py
#!/usr/bin/env python3
"""
Rename `clip_<N>.mp4` → `<N>.mp4` (and `clip_<N>_v<M>.mp4` → `<N>.mp4` keeping
highest version) so assembleEpisode's `^\\d+(\\.\\d+)?\\.mp4$` regex picks them up.

Faithful Python port of normalizeClipFilenames.mjs.

Usage:
  python3 normalizeClipFilenames.py <clips_dir>

Backs up displaced files to <clips_dir>/.originals/ before overwriting.
"""
import re, shutil, sys
from pathlib import Path


def main():
    if len(sys.argv) < 2:
        print("Usage: normalizeClipFilenames.py <clips_dir>", file=sys.stderr)
        sys.exit(1)
    clips_dir = Path(sys.argv[1]).resolve()
    if not clips_dir.is_dir():
        print(f"!! not a directory: {clips_dir}", file=sys.stderr)
        sys.exit(1)

    pat = re.compile(r"^clip_(\d+(?:\.\d+)?)(?:_v(\d+))?\.mp4$")
    groups: dict[str, list[tuple[str, int]]] = {}
    for f in clips_dir.iterdir():
        m = pat.match(f.name)
        if not m: continue
        n = m.group(1)
        v = int(m.group(2)) if m.group(2) else 0
        groups.setdefault(n, []).append((f.name, v))

    if not groups:
        print("(no clip_*.mp4 files to normalize)")
        sys.exit(0)

    backup_dir = clips_dir / ".originals"
    backup_dir.mkdir(exist_ok=True)

    renamed = 0
    for n, lst in groups.items():
        lst.sort(key=lambda x: x[1], reverse=True)   # highest version first
        winner_name, _ = lst[0]
        target = clips_dir / f"{n}.mp4"

        if target.is_file():
            shutil.move(str(target), str(backup_dir / f"{n}.mp4.was"))

        shutil.move(str(clips_dir / winner_name), str(target))
        print(f"  {winner_name}  →  {n}.mp4")
        renamed += 1

        for loser_name, _ in lst[1:]:
            shutil.move(str(clips_dir / loser_name), str(backup_dir / loser_name))
            print(f"    (archived {loser_name})")

    print(f"\n{renamed} clip(s) normalized. Backups in .originals/")


if __name__ == "__main__":
    main()

File: saraandeva/.claude/skills/saraandeva-episode/scripts/_old/normalizeClipFilenames.mjs
#!/usr/bin/env node
// Rename `clip_<N>.mp4` → `<N>.mp4` (and `clip_<N>_v<M>.mp4` → `<N>.mp4`,
// keeping the highest version) so assembleEpisode.mjs's `^\d+(\.\d+)?\.mp4$`
// regex picks them up. Per user preference (integer-only filenames per ep13).
//
// Usage:
//   node normalizeClipFilenames.mjs <clips_dir>
//
// Backs up displaced files to .gitkept_orig/ before overwriting.
import fs from "node:fs";
import path from "node:path";

const dir = path.resolve(process.argv[2] || "");
if (!dir || !fs.existsSync(dir)) {
  console.error("Usage: normalizeClipFilenames.mjs <clips_dir>");
  process.exit(1);
}

const files = fs.readdirSync(dir);
// Group all clip_<N>[_vM].mp4 files by clip number
const groups = {};
for (const f of files) {
  const m = f.match(/^clip_(\d+(?:\.\d+)?)(?:_v(\d+))?\.mp4$/);
  if (!m) continue;
  const n = m[1];
  const v = m[2] ? Number(m[2]) : 0;
  if (!groups[n]) groups[n] = [];
  groups[n].push({ name: f, version: v });
}

const backupDir = path.join(dir, ".originals");
fs.mkdirSync(backupDir, { recursive: true });

let renamed = 0;
for (const [n, list] of Object.entries(groups)) {
  // Pick highest version
  list.sort((a, b) => b.version - a.version);
  const winner = list[0];
  const target = path.join(dir, `${n}.mp4`);

  // Backup any existing target
  if (fs.existsSync(target)) {
    fs.renameSync(target, path.join(backupDir, `${n}.mp4.was`));
  }
  fs.renameSync(path.join(dir, winner.name), target);
  console.log(`  ${winner.name}  →  ${n}.mp4`);
  renamed++;

  // Move non-winning versions to backup
  for (const loser of list.slice(1)) {
    fs.renameSync(path.join(dir, loser.name), path.join(backupDir, loser.name));
    console.log(`    (archived ${loser.name})`);
  }
}

console.log(`\n${renamed} clip(s) normalized. Backups in ${path.relative(dir, backupDir)}/`);
